library/alloc/src/collections/vec_deque.rs

```diff
@@ -1083,24 +1082,16 @@ impl<T> VecDeque<T> {
         self.tail == self.head
     }
 
-    fn range_start_end<R>(&self, range: R) -> (usize, usize)
+    fn range_tail_head<R>(&self, range: R) -> (usize, usize)
     where
         R: RangeBounds<usize>,
     {
-        let len = self.len();
-        let start = match range.start_bound() {
-            Included(&n) => n,
-            Excluded(&n) => n + 1,
-            Unbounded => 0,
-        };
-        let end = match range.end_bound() {
-            Included(&n) => n + 1,
-            Excluded(&n) => n,
-            Unbounded => len,
-        };
-        assert!(start <= end, "lower bound was too large");
-        assert!(end <= len, "upper bound was too large");
-        (start, end)
+        // SAFETY: This buffer is only used to check the range.
+        let buffer = unsafe { slice::from_raw_parts(self.ptr(), self.len()) };
```

Comment:
This is a hack, but it's not too bad.

Reply:
Could this use `buffer_as_slice`?

Reply:
I used that method at first, but I decided to use `from_raw_parts` instead, since it's a hack either way. The issue is that I need a contiguous slice for `check_range`, so I use the first `self.len()` elements even if they're not initialized.

I could change it to `unsafe { self.buffer_as_slice()[..self.len()] }`, but that adds an unnecessary bounds check and looks more valid than it is.

I also added more of this explanation to the safety comment.